sqlobject/src/main/java/org/jdbi/v3/sqlobject/statement/SqlQuery.java

```diff
@@ -58,7 +61,15 @@ public Handler buildHandler(Class<?> sqlObjectType, Method method) {
 
         @Override
         void configureReturner(Query q, SqlObjectStatementConfiguration cfg) {
-            cfg.setReturner(() -> magic.map(q, q.getContext()));
+            cfg.setReturner(() -> {
+                StatementContext ctx = q.getContext();
+                Type elementType = magic.elementType(ctx);
+                UseRowMapper useRowMapper = getMethod().getAnnotation(UseRowMapper.class);
+                ResultIterable<?> iterable = useRowMapper == null
```

Comment:
It's a pity we can't use the `Optional` idiom here because of the type mismatch. I often find very appealing to replace a ternary operator with a series of `Optional` transformations, because they look very clear and elegant. But for such a simple check it's probably not worth it and the C-style code looks better.